### ISSUE TITLE
Not able to execute Java Script on IE11 (Selenium)

### ISSUE THREAD

**Issue by RoheetKamble** [SeleniumHQ/selenium#6966]
While executing the JAVA script 
((JavascriptExecutor) this.webDriver).executeScript("arguments[0].focus();", this.webElement);
on IE 11 i am getting following error.

Error executing JavaScript
Build info: version: '3.12.0',

**Comment by p0deje** [comment]
Please provide full code to reproduce the issue? Please attach all the error backtrace you see.

**Comment by RoheetKamble** [comment]
here i am setting the IE capabilities.

 private WebDriver createIEDriver(final String value, final Timeout driverTimeout) {
        WebDriver driver;
       // System.setProperty("webdriver.ie.driver.loglevel","INFO");  
        String ieDriverPath = System.getProperty("webdriver.ie.driver");
        if(StringUtils.isBlank(ieDriverPath))
        {
        	ieDriverPath =DriverUtils.class.getResource("/IEDriverServer.exe").getPath();
        }
        System.setProperty("webdriver.ie.driver" , ieDriverPath);
        final DesiredCapabilities ieCapabilities = DesiredCapabilities.internetExplorer();
        ieCapabilities.setCapability(InternetExplorerDriver.INTRODUCE_FLAKINESS_BY_IGNORING_SECURITY_DOMAINS,true);
        ieCapabilities.setCapability(InternetExplorerDriver.ENABLE_PERSISTENT_HOVERING,true);
        ieCapabilities.setCapability(InternetExplorerDriver.REQUIRE_WINDOW_FOCUS, true);
        ieCapabilities.setCapability(CapabilityType.ACCEPT_SSL_CERTS, true);
        ieCapabilities.setCapability(InternetExplorerDriver.NATIVE_EVENTS, false);
        ieCapabilities.setPlatform(Platform.WINDOWS);
        ieCapabilities.setCapability("RequireWindowFocus ", false);
        ieCapabilities.setCapability("javascriptEnabled",true);
        ieCapabilities.setCapability("ignoreZoomSetting", true);
        ieCapabilities.setCapability("nativeEvents",false); 
        ieCapabilities.setCapability("unexpectedAlertBehaviour","accept");
        ieCapabilities.setCapability("ignoreProtectedModeSettings",true);
        ieCapabilities.setCapability("disable-popup-blocking",true);
        ieCapabilities.setCapability("PersistentHover",false);
        ieCapabilities.setJavascriptEnabled(true);
        try {
            driverTimeout.start();
            driver = new InternetExplorerDriver(ieCapabilities);
        } catch (final UnhandledException e) {
            maybeKillUnwantedBrowsers(value, DriverUtils.DRIVER_IE);
            driver = new InternetExplorerDriver(ieCapabilities);
        } catch (final WebDriverException e) {
            maybeKillUnwantedBrowsers(value, DriverUtils.DRIVER_IE);
            driver = new InternetExplorerDriver(ieCapabilities);
        } finally {
            try {
                driverTimeout.interrupt();
            } catch (final Exception ex) {
                DriverUtils.LOGGER.debug(ex);
            }
        }
        driver.manage().window().maximize();
        return driver;
    }

and this is my Java script executor command.
((JavascriptExecutor) this.webDriver).executeScript("arguments[0].focus();", this.webElement)

**Comment by RoheetKamble** [comment]
And this error i am getting

org.openqa.selenium.JavascriptException: Error executing JavaScript
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T14:04:26.12Z'
System info: host: 'MUM02L5690', ip: '172.16.27.105', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_112'
Driver info: org.openqa.selenium.ie.InternetExplorerDriver
Capabilities {acceptInsecureCerts: false, browserName: internet explorer, browserVersion: 11, javascriptEnabled: true, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:ieOptions: {browserAttachTimeout: 0, elementScrollBehavior: 0, enablePersistentHover: false, ie.browserCommandLineSwitches: , ie.ensureCleanSession: false, ie.fileUploadDialogTimeout: 3000, ie.forceCreateProcessApi: false, ignoreProtectedModeSettings: true, ignoreZoomSetting: true, initialBrowserUrl: http://localhost:30189/, nativeEvents: true, requireWindowFocus: true}, setWindowRect: true, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: accept}
Session ID: 70efb4f0-d22b-43bd-a60c-2bd445a6819a
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:187)
	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:122)
	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:49)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:158)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:543)
	at org.openqa.selenium.remote.RemoteWebDriver.executeScript(RemoteWebDriver.java:480)
	at com.gep.utils.selenium.PageElement.focus(PageElement.java:151)
	at com.gep.smarttest.selenium.common.Pages.LineItemPage.addLineItem(LineItemPage.java:170)
	at com.gep.smarttest.selenium.common.Pages.LineItemPage.addTabularLineDetails(LineItemPage.java:328)
	at com.gep.smarttest.selenium.fixtures.smartp2p.AddLineItem$exd.execute(AddLineItem.java:47)
	at com.gep.smarttest.spi.StepExecutorClosure.executeStep(StepExecutorClosure.java:161)
	at com.gep.smarttest.spi.StepExecutorClosure.execute(StepExecutorClosure.java:133)
	at org.apache.commons.collections.CollectionUtils.forAllDo(CollectionUtils.java:389)
	at com.gep.smarttest.spi.DefaultSmartTestRunner.executeTestSteps(DefaultSmartTestRunner.java:173)
	at com.gep.mand.FindAndRunSuite.testABC(FindAndRunSuite.java:143)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at components.SmartTestRunnerUI.runTestToGenerateReport(SmartTestRunnerUI.java:435)
	at components.SmartTestRunnerUI.access$000(SmartTestRunnerUI.java:120)
	at components.SmartTestRunnerUI$2.run(SmartTestRunnerUI.java:1143)

**Comment by p0deje** [comment]
Can you give URL to the site where this error can be reproduced?

Also, do you happen to see this error when running in Chrome/Firefox?

**Comment by RoheetKamble** [comment]
Due to some privacy constraints i cant share the URL. For your information same thing works fine on Chrome, i think it must be related to Capabilities issues of IE.

**Comment by p0deje** [comment]
Can you try disable `InternetExplorerDriver.INTRODUCE_FLAKINESS_BY_IGNORING_SECURITY_DOMAINS`?

**Comment by RoheetKamble** [comment]
Hi, appreciating your quick responses. i tried with your mentioned change, still it didn't work; got following error.

Error executing JavaScript
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T14:04:26.12Z'
System info: host: 'MUM02L5690', ip: '172.16.27.105', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_112'
Driver info: org.openqa.selenium.ie.InternetExplorerDriver
Capabilities {acceptInsecureCerts: false, browserName: internet explorer, browserVersion: 11, javascriptEnabled: true, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:ieOptions: {browserAttachTimeout: 0, elementScrollBehavior: 0, enablePersistentHover: false, ie.browserCommandLineSwitches: , ie.ensureCleanSession: false, ie.fileUploadDialogTimeout: 3000, ie.forceCreateProcessApi: false, ignoreProtectedModeSettings: true, ignoreZoomSetting: true, initialBrowserUrl: http://localhost:29392/, nativeEvents: true, requireWindowFocus: true}, setWindowRect: true, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: accept}
Session ID: 09d996bf-793a-4349-9fc5-78e901964824

**Comment by jimevans** [comment]
@RoheetKamble The list of capabilities you’re setting is problematic. There are invalid and even contradictory capabilities being requested. What happens if you remove all of the requested capabilities from your new session request?

**Comment by RoheetKamble** [comment]
The capabilities you see is all about trial and error. Will remove the all capabilities and check again. Will revert soon.Hopefully with positive result ;)

**Comment by RoheetKamble** [comment]
Hi,

I executed the TC with following settings
    private WebDriver createIEDriver(final String value, final Timeout driverTimeout) {
        WebDriver driver;
       // System.setProperty("webdriver.ie.driver.loglevel","INFO");  
        String ieDriverPath = System.getProperty("webdriver.ie.driver");
        System.out.println("System IE driver path:"+ieDriverPath);
        if(StringUtils.isBlank(ieDriverPath))
        {
        	System.out.println("System Driver path is blank, fetching from DriverUtils");
        	ieDriverPath =DriverUtils.class.getResource("/IEDriverServer.exe").getPath();
        	System.out.println("Driver utils IE driver path:"+ieDriverPath);
        }
        System.out.println("IE driver path:"+ieDriverPath);
        System.setProperty("webdriver.ie.driver" , ieDriverPath);
        final DesiredCapabilities ieCapabilities = DesiredCapabilities.internetExplorer();
        
        //Setting Java script enabled
        ieCapabilities.setJavascriptEnabled(true);
        ieCapabilities.setCapability(InternetExplorerDriver.REQUIRE_WINDOW_FOCUS,true);
        
        try {
            driverTimeout.start();
            driver = new InternetExplorerDriver(ieCapabilities);
        } catch (final UnhandledException e) {
            maybeKillUnwantedBrowsers(value, DriverUtils.DRIVER_IE);
            driver = new InternetExplorerDriver(ieCapabilities);
        } catch (final WebDriverException e) {
            maybeKillUnwantedBrowsers(value, DriverUtils.DRIVER_IE);
            driver = new InternetExplorerDriver(ieCapabilities);
        } finally {
            try {
                driverTimeout.interrupt();
            } catch (final Exception ex) {
                DriverUtils.LOGGER.debug(ex);
            }
        }
        driver.manage().window().maximize();
        return driver;
    }

and got following error.

Error executing JavaScript
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T14:04:26.12Z'
System info: host: 'MUM02L5690', ip: '172.16.27.105', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_112'
Driver info: org.openqa.selenium.ie.InternetExplorerDriver
Capabilities {acceptInsecureCerts: false, browserName: internet explorer, browserVersion: 11, javascriptEnabled: true, pageLoadStrategy: normal, platform: WINDOWS, platformName: WINDOWS, proxy: Proxy(), se:ieOptions: {browserAttachTimeout: 0, elementScrollBehavior: 0, enablePersistentHover: false, ie.browserCommandLineSwitches: , ie.ensureCleanSession: false, ie.fileUploadDialogTimeout: 3000, ie.forceCreateProcessApi: false, ignoreProtectedModeSettings: false, ignoreZoomSetting: false, initialBrowserUrl: http://localhost:19351/, nativeEvents: true, requireWindowFocus: true}, setWindowRect: true, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}}
Session ID: e80dc871-d97b-46dc-a8bd-f38c718a1faf

**Comment by jimevans** [comment]
And when you use the latest IE driver (currently 3.141.5)? Beyond that, we’ll really need a reproducible test (including an HTML page or a public URL to test against) to move forward. Executing JavaScript in the IE driver is not globally broken for every user or every page, as the test results at https://webdriver-herald.herokuapp.com show very clearly.

**Comment by barancev** [comment]
Closing the issue because it's not actionable without a reproduction scenario. Feel free to request reopening it as soon as you have got a reproduction case.

**Comment by santanusk** [comment]
I am also getting the same error if the test case is running via Windows Scheduler  and the "run whether user is logged on or not" is checked. The same test case is working fine when I logged in and triggering the application manually. 

We have a .Net application by which we are navigating to our web application and login to check the application is up or not. Our application login page have some javascript\jquery to generate text box for user id and password. I logged the html source in a text file while selenium is opening the page when its been called by windows scheduler, the source does not have those textboxes. But when I debugged the same code in Visual Studio, its properly executing the script and generating the textboxes. 

So it could be a security issue which is blocking javascript when a server is in ideal state. 

I also used Selenium Grid and RemoteDriver but stuck with the same problem. In our case we cant use any other driver except IE. 

Server: Windows Server 2012
IE Driver:  3.141.5 (I placed the exe in application bin folder)
Application : .Net C#
